e2e/test/buttons-tests/buttons-test.js

```diff
@@ -12,20 +12,26 @@ const YEAH_DROPDOWN_NAME = 'Yeah-button';
 const SAVE_BUTTON_NAME = 'Save-button';
 const SAVE_BUTTON_LOADING_NAME = 'Save-button-loading';
 
+let testName = '';
+
 describe('FEF buttons tests', () => {
 
+  beforeEach(function () {
+    testName = this.currentTest.title;
+  });
+
   it('Checks the the fef buttons page title and browser compare visual regression', () => {
     browser.url('/buttons');
     assert.equal(browser.getTitle(), 'Intellyo Application Design System');
-    assert(takeScreenshotAndGetWholePageCompareResult({defaultTolerance: 5.1, ignoreComparison: false, testDirPath: __dirname}), 'Whole FEF buttons page screenshot compare to a reference picture');
+    assert(takeScreenshotAndGetWholePageCompareResult({defaultTolerance: 5.1, ignoreComparison: false, testDirPath: __dirname, testName: testName}), 'Whole FEF buttons page screenshot compare to a reference picture');
```

Comment:
well this is a hard thing, the arrow function. The arrow function inherits the parent's context. The parent in this case is the `describe` which uses the arrow function so it inherits the global which is not something you are looking for i guess.

But you have multiple options here.

1. Using normal function instead of arrow function:

```
it('my function', function() {
   this.test.title === 'my function'; // true
});
```

2. leveraging the mocha setup script

in mocha you can setup special script which gets required when executes your function: https://github.com/webdriverio/wdio-mocha-framework#mochaoptsrequire-stringstring

We use this mock out browser related things in nodejs environment and so on, but you can use to create decorators or global `beforeEach`, eg:

```
// wdio.conf.js
mochaOpts: {
    ui: 'bdd',
    timeout: 99999999,
    require: './e2e/utils/mocha-setup.js'
},
```

```
// ./e2e/utils/mocha-setup.js
beforeEach(function() {  
  browser.currentTestName = this.currentTest.title;
});
```

So in your `debug` statement you can use the `browser.currentTestName` and you don't need to pass that.

With the latter solution, you can still use arrow function.

My suggestion would be using the second one, because having the testname has only debug purposes and nothing else.